tools/lsp/fmt/fmt.rs

```diff
@@ -687,14 +687,13 @@ fn format_repeated_element(
     let mut sub = node.children_with_tokens();
     whitespace_to(&mut sub, SyntaxKind::Identifier, writer, state, "")?;
     whitespace_to(&mut sub, SyntaxKind::DeclaredIdentifier, writer, state, " ")?;
-    // FIXME: [index] should not have a whitespace in front
-    let el = whitespace_to_one_of(
-        &mut sub,
-        &[SyntaxKind::Identifier, SyntaxKind::RepeatedIndex],
-        writer,
-        state,
-        " ",
-    )?;
+
+    let (kind, prefix_whitespace) = if node.child_node(SyntaxKind::RepeatedIndex).is_some() {
+        (SyntaxKind::RepeatedIndex, "")
+    } else {
+        (SyntaxKind::Identifier, " ")
+    };
+    let el = whitespace_to_one_of(&mut sub, &[kind], writer, state, prefix_whitespace)?;
```

Comment:
i guess you can just call whitespace_to here.  In fact i'd do it in the if above.

Can you also check `for [index] in model` without an name?

Reply:
> Can you also check for [index] in model without an name?

Typing that out results in a compiler error with 1.4.0, is this a new syntax?

Reply:
I was thinking it was allowed (i remember wanting this to be allowed while parsing `for` expressions: https://github.com/slint-ui/slint/blob/1d987a45475343a559ccd6ad9d71bb1f4cd2964d/internal/compiler/parser/element.rs#L148 )

But it's actually indeed not accepted, you're right